website/docs/d/regions.html.markdown

```diff
@@ -0,0 +1,68 @@
+---
+subcategory: ""
+layout: "aws"
+page_title: "AWS: aws_regions"
+description: |-
+    Provides list of all enabled AWS regions
+---
+
+# Data Source: aws_regions
+
+`aws_regions` provides list of all enabled AWS regions.
+
+The data source provides list of AWS regions available.
+Can be used to filter regions i.e. by Opt-In status or list only regions enabled for current account.
+To get details like endpoint and description of each region the data source can be combined with `aws_region`.
```

Comment:
In Terraform 0.12 and later, the terminology of `list` versus `set` is meaningful -- list attributes can be indexed while sets cannot.

```suggestion
Provides information about AWS Regions. Can be used to filter regions i.e. by Opt-In status or only regions enabled for current account. To get details like endpoint and description of each region the data source can be combined with the [`aws_region` data source](/docs/providers/aws/d/region.html).
```